eth/vm/forks/frontier/blocks.py

```diff
@@ -136,6 +136,6 @@ def from_header(cls, header: BlockHeaderAPI, chaindb: ChainDatabaseAPI) -> "Fron
     # Execution API
     #
     def add_uncle(self, uncle: BlockHeaderAPI) -> "FrontierBlock":
-        self.uncles.append(uncle)
+        self.uncles += (uncle,)
```

Comment:
Ah, hooray for bugfixes. :) I guess worthy of a news fragment, and surprising and problematic that there's no test coverage on this. Maybe add a test that covers it (or at least an issue to do it as part of beta)? Looks like adding uncles was totally broken. (Is this overwritten in different forks, and that's why we didn't notice?)

Reply:
Good question. I have to investigate that once more tomorrow.

Reply:
Ok, the reason seems to be that this is essentially dead code. At least there's no code in Py-EVM or Trinity calling `add_uncle`. Since this is a public API there's a non-zero chance it is used at another place that I might not be aware of but it seems unlikely. Anyway, I'm not dropping it pre-merge in case I'm overseeing something but I'm happy to send another PR to remove it.

Reply:
> Since this is a public API there's a non-zero chance it is used at another place that I might not be aware of but it seems unlikely.

Yeah, very unlikely, since every call to it would immediately fail :laughing: 

I think we can safely remove an API that has been broken for 3 years:
```
3b0bcc219 eth/vm/forks/frontier/blocks.py   (2019-08-01 11:43:24 -0600 138)     def add_uncle(self, uncle: BlockHeaderAPI) -> "FrontierBlock":
26173f6ed evm/vm/flavors/frontier/blocks.py (2017-06-28 14:50:03 -0600 139)         self.uncles.append(uncle)
26173f6ed evm/vm/flavors/frontier/blocks.py (2017-06-28 14:50:03 -0600 140)         self.header.uncles_hash = keccak(rlp.encode(self.uncles))
26173f6ed evm/vm/flavors/frontier/blocks.py (2017-06-28 14:50:03 -0600 141)         return self
```



> I'm happy to send another PR to remove it.

:+1:

Reply:
Wait, now I'm confused again. The init method adds uncles as `[]`. So I guess maybe it was working before, and we have to think a little more carefully. It's a mess that it's created as a tuple in one place and a list in another.